components/dashboard/src/proceed-with-account.tsx

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import "reflect-metadata";
+import * as React from 'react';
+import * as Cookies from 'js-cookie';
+
+import { createGitpodService } from './service-factory';
+import { ApplicationFrame } from "./components/page-frame";
+import { GitpodHostUrl } from "@gitpod/gitpod-protocol/lib/util/gitpod-host-url";
+
+import Avatar from "@material-ui/core/Avatar";
+import Chip from "@material-ui/core/Chip";
+import Card from "@material-ui/core/Card";
+import CardHeader from "@material-ui/core/CardHeader";
+import CardContent from "@material-ui/core/CardContent";
+import Typography from "@material-ui/core/Typography";
+import { ButtonWithProgress } from "./components/button-with-progress";
+import { themeMode } from './withRoot';
+
+import { renderEntrypoint } from "./entrypoint";
+
+const service = createGitpodService();
+const accountHints = (() => {
+    try {
+        const hints = Cookies.getJSON('ProceedWithAccountCookie');
+        if (accountHints || "currentUser" in hints || "otherUser" in hints) {
+            return hints;
+        }
+    } catch (error) {
+        console.log(error);
+    }
+})();
+
+export class ProceedWithAccount extends React.Component<{}, {}> {
+
+    render() {
+        if (!accountHints) {
+            return (
+                <ApplicationFrame service={service}>
+                    <h3>Oh no! We are missing a cookie here. 🍪</h3>
+                    <h2>A detailed errors message is missing.</h2>
+                </ApplicationFrame>
+            );
+        }
+        const { currentUser, otherUser } = accountHints;
+
+        const accessControlUrl = new GitpodHostUrl(window.location.href).asAccessControl().toString();
+
+        const loginUrl = new GitpodHostUrl(window.location.href).withApi({
+            pathname: '/login/',
+            search: `host=${otherUser.authHost}&returnTo=${encodeURIComponent(accessControlUrl)}`
+        }).toString();
+
+        const logoutUrl = new GitpodHostUrl(window.location.toString()).withApi({
+            pathname: "/logout",
+            search: `returnTo=${encodeURIComponent(loginUrl)}`
+        }).toString();
+
+        const onGoBackClicked = () => {
+            window.location.href = accessControlUrl;
+        };
+        const onSwitchAccountsClicked = () => {
+            window.location.href = logoutUrl;
+        };
+
+        return (
+            <ApplicationFrame service={service}>
+                <div className='content content-area'>
+
+                    <h3>Cannot connect with {otherUser.authHost}</h3>
```

Comment:
**suggestion**: What do you think of the following? We can keep the verbs like switch if you think this describes more accurate what's going on.

| BEFORE | AFTER |
|-|-|
| ![image](https://user-images.githubusercontent.com/120486/106289289-c613ee00-6251-11eb-8014-40cfa3555de0.png) | ![image](https://user-images.githubusercontent.com/120486/106289296-c8764800-6251-11eb-9c1e-6279a6f3df5b.png) |

Reply:
`Created ${days} days ago` and the recommendation on the `Primary account` is postponed. We'd discuss this separately. 

For now we agreed on `Other account`, as we're missing some infos from server to do it.

OTOH, many thanks on the layout work, @gtsiolis! Awesome!